app/addons/components/react-components.react.jsx

```diff
@@ -1081,6 +1105,114 @@ function (app, FauxtonAPI, React, Components, ace, beautifyHelper) {
 
   });
 
+  var ToggleState = React.createClass({
+    getDefaultProps: function () {
+      return {
+        defaultLeft: 'true'
+      };
+    },
+    render: function () {
+      var config = this.props.toggleConfig,
+          defaultLeft = false,
+          defaultRight = false;
+
+      var title = config.title,
+          leftLabel = config.leftLabel,
+          rightLabel = config.rightLabel,
+          leftClick = config.leftClick,
+          rightClick = config.rightClick,
+          enclosingID = config.enclosingID;
+
+      if (config.defaultLeft) {
+        defaultLeft = "checked";
+        defaultRight = false;
+      } else {
+        defaultRight = "checked";
+        defaultLeft = false;
+      }
+
+      return (
+        <div id={enclosingID} className="toggle-states">
+          <div className="toggle-title noselect">{title}</div>
+          <div className="toggles">
+            <input type="radio" 
+              id = {"toggle-state-left-id-" + enclosingID}
```

Comment:
no spaces around the `=`, smame with defaultChecked + a few other attributes in this chunk of code